lib/partitioninfo.coffee

```diff
@@ -57,24 +57,30 @@ getPartitions = (disk, offset, getLogical = true) ->
 					result
 		result
 
-getPartitionFromList = (partitions, number) ->
-	partition = partitions[number - 1]
-	if not partition
-		throw new Error("Partition not found: #{number}.")
-	partition
+partitionNotFoundError = (number) ->
+	throw new Error("Partition not found: #{number}.")
 
-get = (disk, definition) ->
+get = (disk, number) ->
+	if number < 1
+		throw new Error('The partition number must be at least 1.')
 	getPartitions(disk, 0, false)
 	.then (partitions) ->
-		primary = getPartitionFromList(partitions, definition.primary)
-		if not definition.logical
-			primary
-		else if not MBR.Partition.isExtended(primary.type)
-			throw new Error("Not an extended partition: #{definition.primary}.")
+		position = number - 1
+		if partitions.length == 0
+			partitionNotFoundError(number)
+		else if position < partitions.length
+			partitions[position]
 		else
-			getLogicalPartitions(disk, primary.offset, primary.offset, definition.logical - 1)
+			[..., last] = partitions
+			if not MBR.Partition.isExtended(last.type)
```

Comment:
@lurch good catch, I didn't think this was possible.
Only one extended partition is allowed and I was thinking that it had to be the last.
But you're right I just created an image layout like you showed with gparted.
I'll create an issue for this.